python3Packages.fitter, fitter: init at 1.7.1

ShamrockLee opened the pull request:
Package [Fitter](), a Python package and command-line utility for fitting various distributions available under `scipy.stats`.

The upstream test cases cover the library and the executable. Once the package is built successfully, updates and changes should be ready to go.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ShamrockLee committed:
python3Packages.fitter, fitter: init at 1.7.1

Co-authored-by: Fabian Affolter <mail@fabian-affolter.ch>

fabaff commented on pkgs/development/python-modules/fitter/default.nix:
```diff
@@ -0,0 +1,69 @@
+  src = fetchFromGitHub {
+    owner = "cokelaer";
+    repo = "fitter";
+    rev = "v${version}";
```
Helps to avoid branch/tag confusion.

```suggestion
    rev = "refs/tags/v${version}";
```

fabaff commented on pkgs/development/python-modules/fitter/default.nix:
```diff
@@ -0,0 +1,69 @@
+  meta = {
+    description = "Fit data to many distributions";
+    homepage = "https://github.com/cokelaer/fitter";
+    license = lib.licenses.gpl3Only;
```
```suggestion
    changelog = "https://github.com/cokelaer/fitter/releases/tag/v${version}";
    license = lib.licenses.gpl3Only;
```

fabaff commented on pkgs/development/python-modules/fitter/default.nix:
```diff
@@ -0,0 +1,69 @@
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-cov
```
We don't care about the coverage. `pytest-cov-stub` should do the trick.

fabaff commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 355869`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>fitter (python312Packages.fitter)</li>
    <li>fitter.dist (python312Packages.fitter.dist)</li>
    <li>python311Packages.fitter</li>
    <li>python311Packages.fitter.dist</li>
  </ul>
</details>

ShamrockLee commented:
@fabaff, do you think this PR is ready to merge?

ShamrockLee commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 355869`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>fitter (python312Packages.fitter)</li>
    <li>fitter.dist (python312Packages.fitter.dist)</li>
    <li>python311Packages.fitter</li>
    <li>python311Packages.fitter.dist</li>
  </ul>
</details>

ShamrockLee commented:
Merge conflicts resolved.

ShamrockLee commented:
I rebased the feature branch onto the tip of `master` and resolve merge conflicts in `pkgs/top-level/python-packages.nix`.

I also updated fitter from 1.7.1 to 1.8.0.
Since 1.8.0, the upstream adopts NumPy v2 (instead of asserting NumPy v1), so we no longer need to disable runtime dependency checks.

In addition, I also applied the following modernisation:
- Use `finalAttrs` instead of `rec` for overridability.
- Set `__structuredAttrs = true`.

nixpkgs-review-gha commented:
## :x: `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha) ([`b0ab82d`](https://github.com/Defelo/nixpkgs-review-gha/commit/b0ab82d778a1d560a28162740e63d23dad3e546c))

Command: `nixpkgs-review pr 355869`
Commit: [`81357c0a7e78b58d4b0a526eb9aa2091528bf48a`](https://github.com/NixOS/nixpkgs/commit/81357c0a7e78b58d4b0a526eb9aa2091528bf48a) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/81357c0a7e78b58d4b0a526eb9aa2091528bf48a..pull/355869/head))
Merge: [`710c089ef4c34717c9be1a205d121043fc8ebdfd`](https://github.com/NixOS/nixpkgs/commit/710c089ef4c34717c9be1a205d121043fc8ebdfd)

Triggered by @ShamrockLee (44064051)
Logs: https://github.com/ShamrockLee/nixpkgs-review-gha/actions/runs/33052864495/attempts/1



---
### `x86_64-linux`
<details><summary>:white_check_mark: 3 packages built:</summary><ul>
<li>fitter</li>
<li>python313Packages.fitter</li>
<li>python314Packages.fitter</li>
</ul></details>

---
### `aarch64-linux`
<details><summary>:white_check_mark: 3 packages built:</summary><ul>
<li>fitter</li>
<li>python313Packages.fitter</li>
<li>python314Packages.fitter</li>
</ul></details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details><summary>:x: 3 packages failed to build:</summary><ul>
<li>fitter</li>
<li>python313Packages.fitter</li>
<li>python314Packages.fitter</li>
</ul></details>

nixpkgs-review-gha commented:
## :white_check_mark: `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha) ([`b0ab82d`](https://github.com/Defelo/nixpkgs-review-gha/commit/b0ab82d778a1d560a28162740e63d23dad3e546c))

Command: `nixpkgs-review pr 355869`
Commit: [`9715a1235a9d695bfee446942b58d074ce9500b7`](https://github.com/NixOS/nixpkgs/commit/9715a1235a9d695bfee446942b58d074ce9500b7) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/9715a1235a9d695bfee446942b58d074ce9500b7..pull/355869/head))
Merge: [`dbc3362edd05658078630f4b4de2956c199ecf72`](https://github.com/NixOS/nixpkgs/commit/dbc3362edd05658078630f4b4de2956c199ecf72)

Triggered by @ShamrockLee (44064051)
Logs: https://github.com/ShamrockLee/nixpkgs-review-gha/actions/runs/33071707506/attempts/1



---
### `x86_64-linux`
<details><summary>:white_check_mark: 3 packages built:</summary><ul>
<li>fitter</li>
<li>python313Packages.fitter</li>
<li>python314Packages.fitter</li>
</ul></details>

---
### `aarch64-linux`
<details><summary>:white_check_mark: 3 packages built:</summary><ul>
<li>fitter</li>
<li>python313Packages.fitter</li>
<li>python314Packages.fitter</li>
</ul></details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details><summary>:white_check_mark: 3 packages built:</summary><ul>
<li>fitter</li>
<li>python313Packages.fitter</li>
<li>python314Packages.fitter</li>
</ul></details>

ShamrockLee commented:
I used Claude Opus to find the `env.MPLBACKEND = "Agg"` fix and make the tests work on Darwin.
